src/app/wizard/step/provider-settings/provider/extended/component.ts

```diff
@@ -47,20 +50,69 @@ export class ProviderExtendedComponent extends BaseFormValidator implements OnIn
   readonly Providers = NodeProvider;
   readonly Controls = Controls;
 
+  readonly _defaultCIDRAllowAll = '0.0.0.0/0';
+  supportsAllowedIPRangeOverride = false;
+
   constructor(private readonly _builder: FormBuilder, private readonly _clusterSpecService: ClusterSpecService) {
     super('Provider Extended');
   }
 
   ngOnInit(): void {
     this.form = this._builder.group({
       [Controls.ProviderExtended]: this._builder.control(''),
+      [Controls.NodePortsAllowedIPRange]: this._builder.control(this._defaultCIDRAllowAll, [CIDR_PATTERN_VALIDATOR]),
     });
 
     merge(this._clusterSpecService.providerChanges, this._clusterSpecService.datacenterChanges)
       .pipe(takeUntil(this._unsubscribe))
       .subscribe(_ => {
         this.form.removeControl(Controls.ProviderExtended);
         this.form.addControl(Controls.ProviderExtended, this._builder.control(''));
+        this.supportsAllowedIPRangeOverride = this._supportsAllowedIPRangeOverride();
+        if (this.supportsAllowedIPRangeOverride) {
+          this.form.get(Controls.NodePortsAllowedIPRange).addValidators(Validators.required);
+          // Make sure that default value is propagated to cluster spec
+          this._onNodePortsAllowedIPRangeChange(this._defaultCIDRAllowAll);
+        }
       });
+
+    this.form
+      .get(Controls.NodePortsAllowedIPRange)
+      .valueChanges.pipe(takeUntil(this._unsubscribe))
+      .subscribe(this._onNodePortsAllowedIPRangeChange.bind(this));
+  }
+
+  private _onNodePortsAllowedIPRangeChange(value: string): void {
+    switch (this.provider) {
+      case this.Providers.AWS:
```

Comment:
this.Providers is imported there so we can use it in the template, inside the component use the enum directly.